tools/upgrade/steps/upgrade_40_update_resources

```diff
@@ -117,6 +118,22 @@ apply_resources() {
     done
 }
 
+link_pull_secret() {
+  if oc get secret syndesis-pull-secret -o name >/dev/null 2>&1; then
+    # Add pull secrets to every service account specific to syndesis
+    for sa in $(oc get sa -o jsonpath='{.items[*].metadata.name}'); do
```

Comment:
Shouldn't this procedure be performed bny the Operator, rather than the script? Similiar to the inital install, as in https://github.com/syndesisio/syndesis/blob/5972550c6a96e9d04457f0f1ec133474cdab46ed/install/operator/pkg/syndesis/action/install.go#L121 ?

Reply:
yes, also the whole upgrade could be performed by the operator :-) My opinion is that it is safer to go this way when we know that it is working.

And maybe in the future we will have everything in the operator :yum: